scans/tls_scan.py

```
@@ -73,17 +78,19 @@ def _check_cert_valid(self, scan_res: List[ServerScanResult]) -> bool:
     def _get_supported_protocols(self, scan_res: List[ServerScanResult]) -> List[str]:
         protocols = set()
         for res in scan_res:
-            ssl2 = res.scan_commands_results[ScanCommand.SSL_2_0_CIPHER_SUITES]
-            ssl3 = res.scan_commands_results[ScanCommand.SSL_3_0_CIPHER_SUITES]
-            tls1 = res.scan_commands_results[ScanCommand.TLS_1_0_CIPHER_SUITES]
-            tls11 = res.scan_commands_results[ScanCommand.TLS_1_1_CIPHER_SUITES]
-            tls12 = res.scan_commands_results[ScanCommand.TLS_1_2_CIPHER_SUITES]
-            tls13 = res.scan_commands_results[ScanCommand.TLS_1_3_CIPHER_SUITES]
-            protocol_results = [ssl2, ssl3, tls1, tls11, tls12, tls13]
-            for proto in protocol_results:
-                if proto.accepted_cipher_suites:
-                    protocols.add(proto.tls_version_used.name)
 
+            # find accepted cipher suites
+            for cipher in self.CIPHER_SUITES:
+                try:
+                    result = res.scan_commands_results[cipher]
+                    if result.accepted_cipher_suites:
+                        protocols.add(result.tls_version_used.name)
+                except KeyError:
+                    # there was an error trying to get that cipher suite or we did not send the corresponding command
+                    pass
+            # log the scan errors
+            for scan_cmd, err in res.scan_commands_errors.items():
+                logging.error(f"Error when running {scan_cmd}:\n{err.exception_trace}\n")
```

We should `raise` this as an exception if one/many of the SSL/TLS ciphers suites fails to complete otherwise we will hide a failure and continue forward. This would result in a potential false negative case due to failed sslyze scan.